__tests__/app.test.js

```diff
@@ -385,3 +383,78 @@ describe("GET /api/users", () => {
       });
   });
 });
+
+describe('GET /api/articles (topic query)', () => {
+  test('GET: 200 - filters articles by the specified topic', () => {
+    return request(app)
+      .get('/api/articles?topic=cats')
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toBeInstanceOf(Array);
+        body.articles.forEach((article) => {
+          expect(article.topic).toBe('cats');
+        });
+      });
+  });
+  test('GET: 404 - responds with not found for a non-existent topic', () => {
+    return request(app)
+      .get('/api/articles?topic=unknown')
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe('Not Found');
+      });
+  });
+  test('GET: 200 - returns all articles when no topic query is provided', () => {
+    return request(app)
+      .get('/api/articles')
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toBeInstanceOf(Array);
+      });
+  });
+});
+
+describe("GET /api/articles/:article_id (comment_count)", () => {
+  test("GET: 200 - responds with an article object including comment_count", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article).toHaveProperty("author");
```

Comment:
No need ot check the other properties as your part 4 test covers these, but it's all fine other wise!